fedimint-api/src/db/mod.rs

```diff
@@ -392,6 +483,20 @@ impl<'parent> DatabaseTransaction<'parent> {
         }
     }
 
+    pub fn new_module_tx(
+        self,
+        module_instance_id: ModuleInstanceId,
+    ) -> DatabaseTransaction<'parent> {
+        let decoders = self.decoders.clone();
+        let commit_tracker = self.commit_tracker.clone();
+        let wrapped = ModuleDatabaseTransaction::new(self, module_instance_id);
+        DatabaseTransaction {
+            tx: Box::new(wrapped),
+            decoders,
+            commit_tracker,
+        }
+    }
+
```

Comment:
Hmm that's how its coded right now, but I'm up for discussing whether or not they should.

IsolatedDatabaseTransactions can never be committed because they do not own the transaction that can be committed. But this API allows a module (assuming it has access to a Database handle, which it does during ModuleGen), to own a transaction. Right now this is only used in the wallet module and is read-only, so its not a problem, but perhaps other modules will want to write into the database during ModuleGen. What do you guys thinks?

Reply:
Thinking about this a bit more, I think we should let them commit. That's what makes 'ModuleDatabaseTransaction' and 'IsolatedDatabaseTransaction' different. 'IsolatedDatabaseTransaction' needs to be created from outside the module and borrowed when passed into the module. `ModuleDatabaseTransaction` allows a module to own the transaction so it should be able to commit it if it needs to.

I'll add this to the documentation.